tests/validation/tests/v3_api/test_node_template.py

```diff
@@ -0,0 +1,246 @@
+    with pytest.raises(ApiError) as e:
+        do_cloud_credential_config = {"name": "testname3"}
+        user1_client.update(cloud_credential,
+                            digitaloceancredentialConfig=
+                            do_cloud_credential_config)
+    assert e.value.error.status == 403
+    remove_resource(cloud_credential)
+
+
+@if_test_rbac
+def test_rbac_cloud_credential_deploy_cluster(remove_resource):
+    # Admin should be able to use credential to create cluster
+    cloud_credential = create_cloud_credential_do(client)
+    node_template = create_node_template_do(client, cloud_credential)
+    create_and_validate_do_cluster(node_template)
+
+
+# --------------------- helper functions -----------------------
+
+def create_node_template_do(client, cloud_credential=None):
+    if cloud_credential:
+        do_cloud_credential = cloud_credential
+    else:
+        do_cloud_credential_config = {"accessToken": DO_ACCESSKEY}
+        do_cloud_credential = client.create_cloud_credential(
+            digitaloceancredentialConfig=do_cloud_credential_config
+        )
+    node_template = client.create_node_template(
+        digitaloceanConfig={"region": "nyc3",
+                            "size": "2gb",
+                            "image": "ubuntu-18-04-x64"},
+        name=random_name(),
+        driver="digitalocean",
+        cloudCredentialId=do_cloud_credential.id,
+        engineInstallURL=engine_install_url,
+        useInternalIpAddress=True)
+    node_template = client.wait_success(node_template)
+    return node_template
+
+
+def create_cloud_credential_do(client):
+    do_cloud_credential_config = {"accessToken": DO_ACCESSKEY}
+    do_cloud_credential = client.create_cloud_credential(
+        digitaloceancredentialConfig=do_cloud_credential_config
+    )
+    return do_cloud_credential
+
+
+def create_and_validate_do_cluster(node_template,
+                           rancherKubernetesEngineConfig=rke_config,
+                           attemptDelete=True):
+    client = get_admin_client()
+    nodes = []
+    node_name = random_node_name()
+    node = {"hostnamePrefix": node_name,
+            "nodeTemplateId": node_template.id,
+            "controlPlane": True,
+            "etcd": True,
+            "worker": True,
+            "quantity": 1,
+            "clusterId": None}
+    nodes.append(node)
+    cluster, node_pools = create_and_validate_cluster(
+        client, nodes, rancherKubernetesEngineConfig, clusterName=random_name())
+    if attemptDelete:
+        cluster_cleanup(client, cluster)
+    else:
+        return cluster, node_pools
+
+
+@pytest.fixture(autouse="True")
+def create_project_client(request):
+    global client
+    global admin_client
+    global user1_client
+
+    client = get_user_client()
+    admin_client = get_admin_client()
+    user1, user1_token = create_user(admin_client)
+    user1_client = get_client_for_token(user1_token)
+
+    def fin():
+        admin_client.delete(user1)
```

Comment:
Can we add code here to delete the clusters created by the admin users ? If the clusters fail to come up for some reason, they will be deleted as part of this finalizer?

Reply:
This is handled in `create_and_validate_do_cluster`. We call `cluster_cleanup` after the cluster has been created/attempted to be created.